documentation-site/examples/tree-view/controlled.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from 'react';
+import {
+  Unstable_TreeView as TreeView,
+  type TreeNodeT,
+} from 'baseui/tree-view';
+
+const initialData = [
+  {
+    label: 'Node 1',
+    isExpanded: true,
+    children: [
+      {
+        label: 'Child 1',
+        isExpanded: true,
+        children: [
+          {
+            label: 'Grandchild 1',
+          },
+        ],
+      },
+    ],
+  },
+  {
+    label: 'Node 2',
+    isExpanded: true,
+    children: [
+      {
+        label: 'Child 2',
+        isExpanded: true,
+        children: [
+          {
+            label: 'Grandchild 2',
+          },
+        ],
+      },
+    ],
+  },
+];
+
+export default function TreeViewControlled() {
+  const [data, setData] = React.useState(initialData);
+
+  const onToggle = (node: TreeNodeT) => {
+    setData(prevData => {
+      node.isExpanded = !node.isExpanded;
```

Comment:
This feels like it breaks the intuition of React's state update model. The reason why it works is that the node in-scope references an element in the array provided to the component. The approach relies too much on the implementation details of the `TreeView` component and assumes that the reference is always stable. This code will break if the component needs to perform transformations on the array. Can you please refactor these examples to a more immutable approach?